src/cargo/core/features.rs

```diff
@@ -623,6 +623,12 @@ impl Features {
         &self.activated
     }
 
+    /// Temporary workaround for supporting `cargo-features = ["public-dependency"]`.
+    /// FIXME: Need to remove after at least on release cycle.
+    pub fn is_public_dependency_enabled(&self) -> bool {
+        Feature::public_dependency().is_enabled(self)
+    }
```

Comment:
Its not clear to me why this function is needed vs just doing what we did before

Reply:
(generally, when something is short-term, I find it best to be surgical, touching as little as possible)

Reply:
My original idea was that this function would be used to check whether the public-dependency feature was enabled. If not, we can warn it and encourage people to use `-Zpublic-dependency`.

Supporting both approaches seems to be a tricky problem...